src/lib/Form/Type/Language/LanguageChoiceType.php

```diff
@@ -39,9 +44,31 @@ public function getParent()
 
     public function configureOptions(OptionsResolver $resolver)
     {
+        $languageService = $this->languageService;
+        $siteAccessLanguages = $this->configResolver->getParameter('languages');
         $resolver
             ->setDefaults([
-                'choice_loader' => new CallbackChoiceLoader([$this->languageService, 'loadLanguages']),
+                'choice_loader' => new CallbackChoiceLoader(function () use ($languageService, $siteAccessLanguages) {
```

Comment:
As far as I understand a logic behind this, it's a bit overcomplicated. 

How about this?

```php
'choice_loader' => new CallbackChoiceLoader(function() {
    $languages = [];
    $languagesByCode = [];
    
    foreach ($this->languageService->loadLanguages() as $language) {
        $languagesByCode[$language->languageCode] = $language;
    }

    foreach ($this->siteAccessLanguages as $language) {
        if (!isset($languagesByCode[$language]) || !$languagesByCode[$language]->enabled) {
            continue;
        }

        $languages[] = $languagesByCode[$language];
        unset($languagesByCode[$language]);
    }

    return $languages + $languagesByCode;
}),
```

Reply:
Can do it like that, but would probably have to change your last line to something like:
`return array_merge($languages, array_values($languagesByCode));`

ok?

Reply:
@andrerom Sure. :)